python/pyspark/sql/tests.py

```diff
@@ -4931,6 +4931,30 @@ def foo3(key, pdf):
         expected4 = udf3.func((), pdf)
         self.assertPandasEqual(expected4, result4)
 
+    # Regression test for SPARK-24334
+    def test_memory_leak(self):
```

Comment:
Yea, I think it's good enough to have it in PR description. Let's just put this in the PR description.

Reply:
SGTM! Moved to PR description.